### ISSUE TITLE
dependencies for typing-extensions wrongly recorded?

### ISSUE THREAD

**Issue by croth1** [conda-forge/typing_extensions-feedstock#2]
Did something go wrong with the dependencies for the typing-extensions package here?

conda info shows:
```
[...]
dependencies:
    typing_extensions 3.6.6 py27_0
```

Which not only looks strange, but also gives me a solver bug for packages that depend on it:
```
conda.exceptions.UnsatisfiableError: The following specifications were found to be in conflict:
  - python=3.6
  - typing-extensions -> typing_extensions==3.6.5=py27_0 -> python[version='>=2.7,<2.8.0a0'] -> *[track_features=debug]
```

---

conda info typing-extensions

```
typing-extensions 3.6.6 0
-------------------------
file name   : typing-extensions-3.6.6-0.tar.bz2
name        : typing-extensions
version     : 3.6.6
build string: 0
build number: 0
channel     : https://conda.anaconda.org/conda-forge/osx-64
size        : 8 KB
arch        : x86_64
constrains  : ()
license     : PSF
license_family: PSF
md5         : 4f84c79514d31df20258704928f88540
platform    : darwin
subdir      : osx-64
timestamp   : 1538950043837
url         : https://conda.anaconda.org/conda-forge/osx-64/typing-extensions-3.6.6-0.tar.bz2
dependencies:
    typing_extensions 3.6.6 py27_0

```

conda info typing_extensions

```
typing_extensions 3.6.6 py36_0
------------------------------
file name   : typing_extensions-3.6.6-py36_0.tar.bz2
name        : typing_extensions
version     : 3.6.6
build string: py36_0
build number: 0
channel     : https://conda.anaconda.org/conda-forge/osx-64
size        : 31 KB
arch        : x86_64
constrains  : ()
license     : PSF
license_family: PSF
md5         : 9889ac041a94af4f4dab1a4b463e62af
platform    : darwin
subdir      : osx-64
timestamp   : 1538950066771
url         : https://conda.anaconda.org/conda-forge/osx-64/typing_extensions-3.6.6-py36_0.tar.bz2
dependencies:
    python >=3.6,<3.7.0a0
```

**Comment by isuruf** [comment]
For `typing-extensions` you either need to add `python` to host dependencies or loosen the `pin_subpackage` from `exact=True` to the version.
Issue is that without python in `host` dependencies, you are only creating one version of `typing-extensions` and by pinning exactly, you are pinning to one of the python packages of `typing_extensions`

**Comment by croth1** [comment]
Adding python still does not seem to be enough:

```
typing-extensions 3.7.2 1001
----------------------------
file name   : typing-extensions-3.7.2-1001.tar.bz2
name        : typing-extensions
version     : 3.7.2
build string: 1001
build number: 1001
channel     : https://conda.anaconda.org/conda-forge/osx-64
size        : 8 KB
arch        : None
constrains  : ()
license     : PSF
license_family: PSF
md5         : 1ca3bbb939fb8329dc220ccc2cf41b76
platform    : None
subdir      : osx-64
timestamp   : 1552692006022
url         : https://conda.anaconda.org/conda-forge/osx-64/typing-extensions-3.7.2-1001.tar.bz2
dependencies:
    python
    typing_extensions 3.7.2 py27_1001
```

Trying to install this forces a python downgrade to 2.7

**Comment by isuruf** [comment]
You need to add it to host and run

**Comment by croth1** [comment]
Oh sorry, I did not read that carefully.

**Comment by djsutherland** [comment]
Fixed in #15. It was a mistake to add the version with the `-` name, but oh well.
